packages/cdktf-cli/bin/cmds/helper/check-environment.ts

```diff
@@ -0,0 +1,76 @@
+import * as path from "path";
+import * as semver from "semver";
+import { Errors } from "../../../lib/errors";
+import { exec } from "../../../lib/util";
+
+function throwIfLowerVersion(
+  command: string,
+  language: string,
+  minVersion: string,
+  stdout: string
+) {
+  const version = semver.coerce(stdout);
+  if (!version || !semver.valid(version)) {
+    console.error(
+      Errors.Internal(
+        command,
+        `Unable to parse ${language} version "${stdout}"`
+      )
+    );
+    return;
+  }
+
+  if (semver.lt(version, minVersion)) {
+    console.error(
+      Errors.Usage(
+        command,
+        `${language} version "${version}" is not supported. Please upgrade to at least ${minVersion}`
+      )
+    );
+  }
+}
+
+function getBinaryVersion(
+  command: string,
+  binary: string,
+  versionCommand: string
+): Promise<string> {
+  try {
+    return exec(binary, [versionCommand], {});
```

Comment:
Does it make sense to pass down `{ env: process.env }` in case the user configured the env and that might affect how the binary is loaded?

Reply:
yeah right, didn't think of that 👍